### ISSUE TITLE
How to write a test for query_string_parameters

### ISSUE THREAD

**Issue by johansmitsnl** [awslabs/aws-lambda-rust-runtime#252]
I want to write a test for example: https://github.com/awslabs/aws-lambda-rust-runtime/blob/a9de2fcb24030a00e402348aba3c368b717feb6d/lambda-http/examples/hello-http-without-macros.rs#L15-L23

But until now it does not work.

```rust
#[cfg(test)]
mod tests {
    use super::*;
    use lambda_http::http::StatusCode;
    use lambda_http::{http, Body};

    #[tokio::test]
    async fn main_handles() {

        let request = http::Request::builder()
            .uri("https://api.com/?first_name=John")
            .expect("failed to build request");

        let expected = json!({"first_name": "John"}).into_response();
        let response = func(request, Context::default())
            .await
            .expect("expected Ok(_) value")
            .into_response();
        assert_eq!(response.body(), expected.body());
        assert_eq!(response.status(), StatusCode::OK)
    }
}
```

How can I stub the request?

**Comment by softprops** [comment]
You'll want to use [`with_query_string_parameters`](https://github.com/awslabs/aws-lambda-rust-runtime/blob/a9de2fcb24030a00e402348aba3c368b717feb6d/lambda-http/src/ext.rs#L186) should should be accessible in a test config with you import `lambda_http::RequestExt`.

**Comment by mbrubeck** [comment]
`with_query_string_parameters` is not accessible in tests in downstream packages.  It is marked `#[cfg(test)]` so it is included only when building the unit test binary for `lambda_http` itself.  When building tests for downstream crates, `lambda_http` is built as a regular library, without `--cfg test`.

To expose this to downstream crates, it should be compiled unconditionally or behind a Cargo feature.

**Comment by softprops** [comment]
Good to know. If I don't get to it first, we'd be open to a pull request to add that feature flag.

The original intent was to get a small surface area and accomidate use cases that are test specific with a test cfg.

**Comment by johansmitsnl** [comment]
@softprops what would be the code to use it if that flag was enabled?
I run it from github master at the moment.

**Comment by softprops** [comment]
> what would be the code to use it if that flag was enabled?

The target usage is unit tests in consumers crates

**Comment by johansmitsnl** [comment]
I'm relatively new to rust, would it be possible to alter my code above?

**Comment by johansmitsnl** [comment]
This works: #253 in my case. @paulvt helped me finding this change.

If changes are needed please let me know.

**Comment by miere** [comment]
I believe this should be included on any further documentation describing how this _"new lambda runtime"_ will work.

**Comment by schell** [comment]
Where does the `hashmap!` macro come from? This is the first time I've seen that used.

**Comment by paulvt** [comment]
@schell the `maplit` crate, see also: https://github.com/awslabs/aws-lambda-rust-runtime/blob/ae3ce1d4aee593b9425a045cd19d2b967d5cf367/lambda-http/src/lib.rs#L90

**Comment by johansmitsnl** [comment]
This is now solved in #253
